data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/transformer/DynamicConfigTransformer.java

```diff
@@ -402,15 +403,28 @@ private boolean isJsonPath(String parameter) {
     }
 
     /**
-     * Specific to DocDB depth field.
-     * @param s3Prefix
-     * @return
+     * Calculate s3 folder scan depth for DocDB source pipeline
+     * @param s3Prefix: s3 prefix defined in the source configuration
+     * @return s3 folder scan depth
      */
     public String calculateDepth(String s3Prefix) {
+        return Integer.toString(getDepth(s3Prefix, 4));
+    }
+
+    /**
+     * Calculate s3 folder scan depth for RDS source pipeline
+     * @param s3Prefix: s3 prefix defined in the source configuration
+     * @return s3 folder scan depth
+     */
+    public String calculateDepthForRdsSource(String s3Prefix) {
+        return Integer.toString(getDepth(s3Prefix, 3));
```

Comment:
Why is the difference between RDS and DocDB ?

Reply:
I removed the timestamp in EpochMillis from S3 path, so it is one layer shorter compared to DocumentDB.

Reply:
Discussed offline. Will make it consistent with DocDB.

Reply:
Discussed again. The S3 prefix looks like this: `user defined prefix / source identifier / buffer / folder partitions`. So I have depth = length of user defined prefix + 3.